provider/internal/java/dependency.go

```diff
@@ -336,22 +350,47 @@ func (p *javaServiceClient) parseDepString(dep, localRepoPath string) (provider.
 	// Split string on ":" must have 5 parts.
 	// For now we ignore Type as it appears most everything is a jar
 	parts := strings.Split(dep, ":")
-	if len(parts) != 5 {
-		return d, fmt.Errorf("unable to split dependency string %s", dep)
+	if len(parts) >= 3 {
+		// Its always <groupId>:<artifactId>:<Packaging>: ... then
+		d.Type = parts[2]
+		if len(parts) == 6 {
+			// parts[5] is the <scope>
+			// parts[3] is the <classifier>
+			d.Version = parts[4]
+		} else if len(parts) == 5 {
+			// parts[4] is the <scope>
+			d.Version = parts[3]
+		} else {
+			p.log.Info("Cannot derive version from dependency string", "dependency", dep)
+			d.Version = "Unknown"
+		}
+	} else {
+		p.log.Info("unable to analyze dependency", "dependency", dep)
+		d.Name = "Unknown"
+		d.Version = "Unknown"
+		d.Type = "Unknown"
+		d.Labels = []string{""}
+		d.FileURIPrefix = ""
+		d.ResolvedIdentifier = ""
+
+		return d, nil
 	}
 	d.Name = fmt.Sprintf("%s.%s", parts[0], parts[1])
-	d.Version = parts[3]
-	d.Type = parts[4]
 
 	fp := filepath.Join(localRepoPath, strings.Replace(parts[0], ".", "/", -1), parts[1], d.Version, fmt.Sprintf("%v-%v.jar.sha1", parts[1], d.Version))
-	b, err := os.ReadFile(fp)
-	if err != nil {
-		// Log the error and continue with the next dependency.
-		p.log.V(5).Error(err, "error reading SHA hash file for dependency", "dep", d.Name)
-		// Set some default or empty resolved identifier for the dependency.
+	if _, err := os.Stat(fp); os.IsNotExist(err) {
```

Comment:
Will ReadFile, give you the same error, or does it give a different error?

Reply:
?? ReadFile returns an error if the file does not exist. Therefore, I consider it a good practice in programming to check whether the file exists, before trying to open it.

Reply:
If that is the case, I would leave this logic as is. 

We get the same error, and it goes to the log explaining that the file does not exist. And we remove complexity to only have a single if/else block.

Reply:
Seems we have different opinions here. But may I remind you of the _good-citizen_ practice mentioned elsewhere? ;-)

Reply:
Please see the library function ReadFile:
 https://cs.opensource.google/go/go/+/refs/tags/go1.21.3:src/os/file.go;drc=79a8997ac5592f11388d9f26ab3fff65ec876dbd;l=714

Reply:
I think in golang the typical practice would be to attempt to read the file and check the error for failure. 

As far as I can tell there are no differences between the approaches other than added handling complexity and an extra syscall if we check for existence first. The ReadFile function is built with the expectation that it may be called on non-existent files and has no unpleasant side-effects. 

Since we log the error anyway it will be clear the reason the file read failed without the extra check

Reply:
I think what shawn's pointing at is that - it's redundant to check for file exists, as the `err` variable in ReadFile will capture the same error itself.

Reply:
oops, sorry, fabian and I responded at the same time, +1

Reply:
Will not push this anymore. Nevertheless, an error is logged for missing .sha1 files which might not be in the local repo for my own modules. What bothers me here, is that this is logged as error. Errors imply that I (as a user/developer/ops person) have done something wrong. And there is no fault in this case. So for my own modules I would like to see the message logged in maximum as a warning, or even only info or lower (debug).

Reply:
I think I agree with this and would plus one logging this as a V(5).Info while including the error as a key-value pair. You bring up a very good point about it being an error.